File: elixir/space-age/space_age.exs
defmodule SpaceAge do
  @type planet :: :mercury | :venus | :earth | :mars | :jupiter
  | :saturn | :neptune | :uranus

  @earth_year_seconds 31557600
  @planet_year_seconds %{
    mercury: 0.2408467 * @earth_year_seconds,
    venus: 0.61519726 * @earth_year_seconds,
    earth: @earth_year_seconds,
    mars: 1.8808158 * @earth_year_seconds,
    jupiter: 11.862615 * @earth_year_seconds,
    saturn: 29.447498 * @earth_year_seconds,
    uranus: 84.016846 * @earth_year_seconds,
    neptune: 164.79132 * @earth_year_seconds
  }

  @doc """
  Return the number of years a person that has lived for 'seconds' seconds is
  aged on 'planet'.
  """
  @spec age_on(planet, pos_integer) :: float
  def age_on(planet, seconds) do
    seconds / @planet_year_seconds[planet]
  end
end

File: erlang/space-age/space_age.erl
-module(space_age).
-export([ageOn/2]).

-define(PlanetYearsInEarthYears,
        [
         {mercury, 0.2408467},
         {venus, 0.61519726},
         {earth, 1},
         {mars, 1.8808158},
         {jupiter, 11.862615},
         {saturn, 29.447498},
         {uranus, 84.016846},
         {neptune, 164.79132}
        ]).

-define(PlanetYearsInSeconds, 
        lists:map(fun ({P, Y}) -> {P, Y * 31557600} end, 
                  ?PlanetYearsInEarthYears)).

ageOn(Planet, Seconds) ->
    Seconds / proplists:get_value(Planet, ?PlanetYearsInSeconds).
